packages/core/src/types/aggregator/aggregator.interface.ts

```diff
@@ -0,0 +1,23 @@
+import { BlockTransactionCommitment, Transaction } from '../serialization'
+
+export interface Aggregator {
+  /**
+   * Notifies the Aggregator of the provided Transaction so it may be included in the next block.
+   *
+   * @param transaction the Transaction in question
+   * @param witness the Witness of the Transaction in question
+   *
+   * @returns the BlockTransactionCommitment indicating the transaction will be included in the next block
+   */
+  ingestTransaction(
+    transaction: Transaction,
+    witness: string
+  ): Promise<BlockTransactionCommitment>
```

Comment:
It is unclear to me what the `witness` parameter does here. I believe it may have been part of our out of date spec.

Ideally we'd simply be sent the transaction itself. From this we run some authentication logic & generate a new StateUpdate. In fact this whole transaction ingestion process is under specified and so I just created an issue for it here: https://github.com/plasma-group/pigi/issues/349

Reply:
That's a good question. There is both authentication and authorization to be done, right?

1. Is this transaction actually from the user who is trying to send their funds?
2. Does the user trying to send funds actually own the funds they're trying to send?

I assumed the witness portion would solve (1), although that could be handled in the communication layer itself. Is there something I'm missing that either handles that or makes it unnecessary?

Reply:
I think that we can keep transactions more general and with that we won't need to break those two (authentication/authorization) into two steps. Instead you simple check that the transaction itself is valid. The tx might not even have a single sender--it could simply be a subsidized function call in a smart contract (essentially the concept of account abstraction https://github.com/ethereum/EIPs/issues/859).

> although that could be handled in the communication layer itself
Yeah I think it makes sense for the anti-spam mechanism to be handled by the communication layer. Eg. banning peers that repeatedly broadcast failing transactions.

Reply:
Thanks for the explanation 😄. I'll remove witness from the method signature.